runtime/compiler/control/CompilationThread.cpp

```diff
@@ -2008,7 +2008,7 @@ bool TR::CompilationInfo::shouldRetryCompilation(TR_MethodToBeCompiled *entry, T
    if (entry->_compErrCode != compilationOK &&
       useSeparateCompilationThread()) // Retrials for compilation on application thread are not implemented
       {
-      if (entry->_compilationAttemptsLeft > 0)
+      if ((entry->_compilationAttemptsLeft > 0) && TR::Options::canJITCompile())
```

Comment:
Suggestion: I see this pattern repeating a few times `(entry->_compilationAttemptsLeft > 0) && TR::Options::canJITCompile()`. This can be simplified if we always set entry->_compilationAttemptsLeft to 1 if TR::Options::canJITCompile() is false.